From dff2d21c01001648c9265b11cb410e1d3bfb10f9 Mon Sep 17 00:00:00 2001
From: Vitaly Andrianov <vitalya@ti.com>
Date: Wed, 3 Jan 2018 14:52:15 -0500
Subject: [PATCH] prueth: fix lre_cnt_nodes address

This patch adds back the the lre_cnt_nodes variable to fix the snmp
query handling for Node Table

Signed-off-by: Vitaly Andrianov <vitalya@ti.com>
Signed-off-by: Murali Karicheri <m-karicheri2@ti.com>
---
 drivers/net/ethernet/ti/pruss_node_tbl.c | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/drivers/net/ethernet/ti/pruss_node_tbl.c b/drivers/net/ethernet/ti/pruss_node_tbl.c
index 15e11af..071cf63 100644
--- a/drivers/net/ethernet/ti/pruss_node_tbl.c
+++ b/drivers/net/ethernet/ti/pruss_node_tbl.c
@@ -15,6 +15,7 @@
 #include <linux/string.h>
 #include <linux/spinlock_types.h>
 #include "pruss_node_tbl.h"
+#include "hsr_prp_firmware.h"
 
 #define IND_BINOFS(x) nt->index_tbl[x].bin_offset
 #define IND_BIN_NO(x) nt->index_tbl[x].bin_no_entries
@@ -77,21 +78,24 @@ int node_table_insert(struct prueth *prueth, u8 *mac, int port, int sv_frame,
 
 static inline void lre_cnt_nodes_clear(struct node_tbl *nt)
 {
-	u32 *lre_cnt_nodes = (u32 *)((void *)nt + 192 - 0x3000);
+	u32 *lre_cnt_nodes = (u32 *)((void *)nt + LRE_CNT_NODES -
+				     NODE_TABLE_NEW);
 	*lre_cnt_nodes = 0;
 	nt->lre_cnt = 0;
 }
 
 static inline void lre_cnt_nodes_inc(struct node_tbl *nt)
 {
-	u32 *lre_cnt_nodes = (u32 *)((void *)nt + 192 - 0x3000);
+	u32 *lre_cnt_nodes = (u32 *)((void *)nt + LRE_CNT_NODES -
+				     NODE_TABLE_NEW);
 	*lre_cnt_nodes += 1;
 	nt->lre_cnt++;
 }
 
 static inline void lre_cnt_nodes_dec(struct node_tbl *nt)
 {
-	u32 *lre_cnt_nodes = (u32 *)((void *)nt + 192 - 0x3000);
+	u32 *lre_cnt_nodes = (u32 *)((void *)nt + LRE_CNT_NODES -
+				     NODE_TABLE_NEW);
 	*lre_cnt_nodes -= 1;
 	nt->lre_cnt--;
 }
-- 
1.9.1

